### PR TITLE
remove Failover from scheduler

### LINKED ISSUE #1762: TODO List: Enhance Cluster Lifecycle Management
**What would you like to be added**:
According to https://github.com/karmada-io/karmada/issues/1640#issuecomment-1119377312, I'd like to enhance cluster lifecycle management. Here is the todo-list.

- [x] Add a taint for the clusters which are unhealthy for a period of time.#1781
- [x] Add taint manager. #1945
- [x] Remove cluster scheduling result in `spec.clusters` when the cluster is terminating. #2373
- [ ] Remove `Failover` from scheduler. #2391

**Why is this needed**:
Move some functions away from scheduler and make the scheduler more concise.

### PR DESCRIPTION
**Pull request by Garrybest**
Signed-off-by: Garrybest <garrybest@foxmail.com>

**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #1762

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Remove Failover from scheduler
```

### CODE REVIEW

**Comment by XiShanYongYe-Chang** [comment]
Thanks~
/assign

**Comment by Garrybest** [comment]
Refer to https://github.com/karmada-io/karmada/pull/2301#issuecomment-1219554645.

**Comment by XiShanYongYe-Chang** [comment]
It seems we need to adjust the E2E of failover.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
/lgtm
/cc @RainbowMango

File: pkg/scheduler/event_handler.go [hunk 1]
```diff
@@ -195,48 +193,6 @@ func (s *Scheduler) addCluster(obj interface{}) {
...
-		clusters := binding.Spec.Clusters
-		for _, bindingCluster := range clusters {
-			if bindingCluster.Name == notReadyClusterName {
-				rescheduleKey, err := cache.MetaNamespaceKeyFunc(binding)
-				if err != nil {
-					klog.Errorf("couldn't get rescheduleKey for ResourceBinding %#v: %v", bindingCluster.Name, err)
-					return
-				}
-				s.queue.Add(rescheduleKey)
-				metrics.CountSchedulerBindings(metrics.ClusterNotReady)
```
Comment (RainbowMango) [inline pkg/scheduler/event_handler.go:212]: Should we clean up the metrics `ClusterNotReady` which is not used anymore?

Comment (Garrybest) [reply pkg/scheduler/event_handler.go:212]: +1

**Review by RainbowMango — APPROVED** [review]
/lgtm
/approve
Thanks.

**Pull request merged**